eZ/Bundle/EzPublishCoreBundle/Command/ReindexCommand.php

```diff
@@ -0,0 +1,178 @@
+            ->setDescription('Indexes the configured database for configured search engine')
+            ->addArgument('bulk_count', InputArgument::OPTIONAL, 'Number of Content objects indexed at once', 5)
+            ->setHelp(
+                <<<EOT
+The command <info>%command.name%</info> indexes content objects for configured search engine.
+EOT
+            );
+    }
+
+    protected function execute(InputInterface $input, OutputInterface $output)
+    {
+        $this->logger = $this->getContainer()->get('logger');
+
+        $bulkCount = $input->getArgument('bulk_count');
+
+        /** @var \eZ\Publish\SPI\Search\Handler $searchHandler */
+        $searchHandler = $this->getContainer()->get('ezpublish.spi.search');
+        if (!$searchHandler instanceof SearchHandlerInterface) {
+            throw new RuntimeException(
+                'Expected to find Search Handler but found something else.'
+            );
+        }
+
+        /* @var \eZ\Publish\SPI\Persistence\Handler $persistenceHandler */
+        $persistenceHandler = $this->getContainer()->get('ezpublish.api.persistence_handler');
+        /* @var \eZ\Publish\Core\Persistence\Database\DatabaseHandler $databaseHandler */
+        $databaseHandler = $this->getContainer()->get('ezpublish.connection');
+
+        // Indexing Content
+        $totalCount = $this->getContentObjectsTotalCount($databaseHandler, ContentInfo::STATUS_PUBLISHED);
+
+        $query = $databaseHandler->createSelectQuery();
+        $query->select('id', 'current_version')
+            ->from('ezcontentobject')
+            ->where($query->expr->eq('status', ContentInfo::STATUS_PUBLISHED));
+
+        $stmt = $query->prepare();
+        $stmt->execute();
+
+        /* @var \eZ\Publish\SPI\Search\Handler $searchHandler */
+        $searchHandler->purgeIndex();
+
+        $output->writeln('Indexing Content...');
+
+        /* @var \Symfony\Component\Console\Helper\ProgressHelper $progress */
+        $progress = $this->getHelperSet()->get('progress');
+        $progress->start($output, $totalCount);
+        $i = 0;
+        do {
+            $contentObjects = array();
+
+            for ($k = 0; $k <= $bulkCount; ++$k) {
+                if (!$row = $stmt->fetch(PDO::FETCH_ASSOC)) {
+                    break;
+                }
+                try {
+                    $contentObjects[] = $persistenceHandler->contentHandler()->load(
+                        $row['id'],
+                        $row['current_version']
+                    );
+                } catch (NotFoundException $e) {
+                    $progress->clear();
+                    $this->logWarning($output, "Could not load current version of Content with id ${row['id']}, so skipped for indexing. Full exception: " . $e->getMessage());
+                    $progress->display();
+                }
+            }
+
+            $documents = $this->generateDocuments($contentObjects, $searchHandler, $progress, $output);
+
```

Comment:
Ok. Anyway, as in SolrCreateIndexCommand, I changed here `generateDocuments` to inline block.